modules/create_timepoint/php/timepoint.class.inc

```diff
@@ -0,0 +1,348 @@
+        $config         = \NDB_Config::singleton();
+        $candidate      = \Candidate::singleton(new CandID($values['candID']));
+        $allSubprojects = \Utility::getSubprojectList();
+
+        // All subprojects from config file (error).
+        if (empty($allSubprojects)) {
+            $conflict['subprojectID'] = 'No subprojects have been defined
+            for this study. If you are an administrator, please use the
+            Configuration module to add new subprojects.';
+        }
+
+        // List of valid subprojects for a given project
+        $subprojList = $candidate->getValidSubprojects();
+        if (empty($subprojList)) {
+            $conflict['subprojectID'] = 'No subprojects have been
+            defined for the project this candidate is affiliated with.
+            If you are an administrator, please use the Configuration module to
+            add new subprojects and associate them with projects.';
+        }
+        $values['subproject'] = array_intersect_key(
+            \Utility::getSubprojectList(),
+            $candidate->getValidSubprojects()
+        );
+
+        // Retrieve visit labels.
+        $visit_options      = [];
+        $visitLabelSettings = $config->getSetting('visitLabel');
+        foreach (
+            \Utility::associativeToNumericArray($visitLabelSettings) as $visitLabel
+        ) {
+            if (!empty($values['subproject'])) {
+                $visitLabel   = \Utility::associativeToNumericArray($visitLabel)[0];
+                $labelOptions = [];
+                $items        = \Utility::associativeToNumericArray(
+                    $visitLabel['labelSet']['item']
+                );
+                foreach ($items as $item) {
+                    $item = \Utility::associativeToNumericArray($item)[0];
+                    $labelOptions[$item['@']['value']] = $item['#'];
+                }
+                $visit_options[$visitLabel['@']['subprojectID']]
+                    = array_filter($labelOptions);
+            }
+        }
+        $values['visit'] = $visit_options;
+
+        // List sites
+        $values['psc'] = array_intersect(
+            \Utility::getSiteList(),
+            \User::singleton()->getSiteNames()
+        );
+
+        // List languages
+        $languages = \Utility::getLanguageList();
+        if (count($languages) > 1) {
+            array_unshift($languages, null);
+        }
+        $values['languages'] = $languages;
+
+        // List projects
+        $user = \User::singleton();
+        $user_list_of_projects = $user->getProjectIDs();
+        if (count($user_list_of_projects) > 1) {
+            $projectOptions = [];
+            foreach ($user_list_of_projects as $projectID) {
+                $project = \Project::getProjectFromID($projectID);
+                $projectOptions[$projectID] = $project->getName();
+            }
+            $values['project'] = $projectOptions;
+        }
```

Comment:
the project isn't showing up . i think you should remove the > 1 check so that the project just gets listed, even though there's only 1, similar to 'list sites'

Reply:
@zaliqarosli updated thanks